src/Microsoft.AspNetCore.Blazor.Browser/Rendering/BrowserRenderer.cs

```diff
@@ -78,10 +85,19 @@ public void Dispose()
         /// <inheritdoc />
         protected override void UpdateDisplay(in RenderBatch batch)
         {
-            RegisteredFunction.InvokeUnmarshalled<int, RenderBatch, object>(
-                "renderBatch",
-                _browserRendererId,
-                batch);
+            if (JSRuntime.Current is MonoWebAssemblyJSRuntime mono)
+            {
+                mono.InvokeUnmarshalled<int, RenderBatch, object>(
+                    "Blazor._internal.renderBatch",
+                    _browserRendererId,
+                    batch);
+            }
+            else
+            {
+                // When implementing support for an out-of-process JS runtime, we'll need to
+                // do something here to serialize and transmit the RenderBatch efficiently.
+                throw new NotImplementedException("TODO: Support BrowserRenderer.UpdateDisplay on other runtimes.");
```

Comment:
Yes, that's possible! I strongly expect we'll be tweaking the layering a bit when/if we implement another Renderer, since there is probably some aspects of `BrowserRenderer` that we'd want to share across renderers. The exact layer boundaries probably have to move around a bit.